sdk/core/azure-core/src/http/retry_policy.cpp

```diff
@@ -246,6 +227,7 @@ std::unique_ptr<RawResponse> RetryPolicy::Send(
     request.GetUrl().SetQueryParameters(std::move(originalQueryParameters));
 
     // Update retry number
-    retryContext = CreateRetryContext(retryContext);
+    // retryContext = CreateRetryContext(retryContext);
```

Comment:
remove commented out code

Reply:
Don't we need to update `retryContext` in each iteration of the loop before passing it to `Send`?

Reply:
No, instead of updating the context, we make the context to keep a pointer to the value that we update.
Then, the context is acting as a pointer to the value that gets updated :)